qiskit_dynamics/pulse/pulse_to_signals.py

```diff
@@ -79,14 +81,60 @@ def __init__(
         self._carriers = carriers or {}
 
     def get_signals(self, schedule: Schedule) -> List[DiscreteSignal]:
-        """
+        r"""Convert a schedule to a corresponding list of DiscreteSignal instances.
+
+        Which channels are converted, and the order they are returned, is controlled by the
+        ``channels`` argument at instantiation. The ``carriers`` instantiation argument sets the
+        analog carrier frequency for each channel, which is fixed for the full duration. For a given
+        channel, the :math:`k^{th}` envelope sample for the corresponding :class:`.DiscreteSignal`
+        is determined according to the following formula:
+
+        .. math::
+            f_k \exp(i(2\pi \Delta\nu(k) k dt + \phi(k) + 2 \pi \phi_a(k))),
+
+        where:
+
+        * :math:`f_k` is the waveform value at the :math:`k^{th}` time step as specified by ``Play``
+          instructions.
+        * :math:`\Delta\nu(k)` is the frequency deviation at time step :math:`k` from the analog
+          carrier as the result of ``SetFrequency`` and ``ShiftFrequency`` instructions. As evident
+          by the formula, carrier frequency deviations as a result of these instructions are handled
+          digitally, with the analog carrier frequency being fixed for the entirety of the schedule.
+        * :math:`dt` is the sample rate as specified by the ``dt`` instantiation argument.
+        * :math:`\phi(k)` is the channel phase at time step :math:`k`, as determined by
+          ``ShiftPhase`` and ``SetPhase`` instructions.
+        * :math:`\phi_a(k)` is the phase correction term at time step :math:`k`, impacted by
+          ``SetFrequency`` and ``ShiftFrequency`` instructions, described below.
```

Comment:
Does it say somewhere which \phi comes from the pulse profile? (rather than phase advancement instructions, etc)

Reply:
The `\phi` is purely determined by `ShiftPhase` and `SetPhase`, and `phi_a` is purely determined by `SetFrequency` and `ShiftFrequency` instructions. This is the only way to manipulate these parameters. The `Play` instructions themselves have no phase, they just use whatever the current channel phase is.

Reply:
Oh so f_k is complex-valued. Got it.

Reply:
Ah I see what you're saying, and why this might be confusing then, though this is the convention used in qiskit pulse.